src/log4net/DateFormatter/AbsoluteTimeDateFormatter.cs

```diff
@@ -95,78 +87,52 @@ protected virtual void FormatDateWithoutMillis(DateTime dateToFormat, StringBuil
     /// per second.
     /// </para>
     /// <para>
-    /// Sub classes should override <see cref="FormatDateWithoutMillis"/>
+    /// Subclasses should override <see cref="FormatDateWithoutMillis"/>
     /// rather than <see cref="FormatDate"/>.
     /// </para>
     /// </remarks>
     public virtual void FormatDate(DateTime dateToFormat, TextWriter writer)
     {
-      lock (s_lastTimeStrings)
-      {
-        // Calculate the current time precise only to the second
-        long currentTimeToTheSecond = (dateToFormat.Ticks - (dateToFormat.Ticks % TimeSpan.TicksPerSecond));
-
-        string timeString = null;
-        // Compare this time with the stored last time
-        // If we are in the same second then append
-        // the previously calculated time string
-        if (s_lastTimeToTheSecond != currentTimeToTheSecond)
-        {
-          s_lastTimeStrings.Clear();
-        }
-        else
-        {
-          timeString = (string)s_lastTimeStrings[GetType()];
-        }
-
-        if (timeString == null)
+      string timeString = s_lastTimeStrings.AddOrUpdate(GetType(),
```

Comment:
Can you please add some tests?
The whole class seems to be uncovered.

Reply:
There was a class already but it was explicitly excluded from compilation. Restored and added more coverage.